.github/templates/windows_ci_workflow.yml.j2

```diff
@@ -118,7 +118,6 @@ jobs:
           .jenkins/pytorch/win-build.sh
       # Upload to github so that people can click and download artifacts
       - name: Upload artifacts to Github
```

Comment:
Do we even want this step anymore? I notice we don't upload artifacts to GitHub on Linux

Reply:
I think we can remove it, the gha artifact store is pretty redundant with S3 and worse in terms of speed